components/Input/index.jsx

```diff
@@ -1,17 +1,25 @@
-import React, {PropTypes} from 'react';
+import React, {Component, PropTypes} from 'react';
 import classnames from 'classnames';
 import {
   modifierClassList, defaultReactProps, defaultReactPropsValues
 } from '../utils';
 
-const Input = ({className, ...props}) => {
-  let {classList, ...finalProps} = modifierClassList(props);
-  classList = classnames('input', className, classList);
+class Input extends Component {
+  constructor(props) {
+    super(props);
 
-  return (
-    <input className={classList} {...finalProps} />
-  );
-};
+    let {classList, ...finalProps} = modifierClassList(props);
```

Comment:
I wouldn't be placing these in the constructor. You should be able to pass in `{...finalProps}` and just set classnames to a var.